File: src/main/java/com/iotassistant/models/notifications/DeviceOfflineNotification.java
package com.iotassistant.models.notifications;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.iotassistant.models.devices.Device;
import com.iotassistant.models.devices.WatchdogInterval;

@Entity
public class DeviceOfflineNotification extends Notification{
	
	@NotNull
	@ManyToOne
	@OnDelete(action = OnDeleteAction.CASCADE)
	private Device device;
	
	private String watchdogInterval;

	public DeviceOfflineNotification() {
		super();
	}

	public DeviceOfflineNotification(Device device, String date, WatchdogInterval watchdogInterval) {
		super(date);
		this.device = device;
		this.watchdogInterval = watchdogInterval.toString();
	}

	public String getDeviceName() {
		return device.getName();
	}
	
	@Override
	public void accept(NotificationVisitor notificationVisitor) {
		notificationVisitor.visit(this);		
	}

	public String getWatchdogInterval() {
		return watchdogInterval;
	}
	
	

}

File: src/main/resources/static/app/notifications/entities/DeviceOfflineNotification.js
class DeviceOfflineNotification extends Notification{
		
	constructor(id, date, deviceName, watchdogInterval) {
		super(id, date);
		this.deviceName = deviceName;
		this.watchdogInterval = watchdogInterval;
	}
	
	toString() {
		var deviceName = this.deviceName;
		var text = "Component " + deviceName + " went offline" + ". Date: " + this.date + ", watchddog interval: " + this.watchdogInterval;
		return text;
	}
	
	
}